backend/src/apiserver/resource/resource_manager.go

```diff
@@ -195,616 +203,902 @@ func (r *ResourceManager) ArchiveExperiment(ctx context.Context, experimentId st
+	if err != nil {
+		err = util.Wrapf(err, "Failed to list pipelines with context %v, options %v", filterContext, opts)
+	}
+	return pipelines, total_size, nextPageToken, err
+}
+
+// TODO(gkcalat): consider removing after KFP v2 GA if users are not affected.
+// Returns a list of pipelines using LEFT JOIN on SQL query.
+// This could be more performant for a large number of pipeline versions.
+func (r *ResourceManager) ListPipelinesV1(filterContext *model.FilterContext, opts *list.Options) ([]*model.Pipeline, []*model.PipelineVersion, int, string, error) {
+	pipelines, pipelineVersions, total_size, nextPageToken, err := r.pipelineStore.ListPipelinesV1(filterContext, opts)
+	if err != nil {
+		err = util.Wrapf(err, "ResourceManager (v1beta1): Failed to list pipelines with context %v, options %v", filterContext, opts)
+	}
+	return pipelines, pipelineVersions, total_size, nextPageToken, err
 }
 
+// Returns a pipeline.
 func (r *ResourceManager) GetPipeline(pipelineId string) (*model.Pipeline, error) {
-	return r.pipelineStore.GetPipeline(pipelineId)
+	if pipeline, err := r.pipelineStore.GetPipeline(pipelineId); err != nil {
+		return nil, util.Wrapf(err, "Failed to get a pipeline with id %v", pipelineId)
+	} else {
+		return pipeline, nil
+	}
 }
 
+// Returns a pipeline specified by name and namespace.
 func (r *ResourceManager) GetPipelineByNameAndNamespace(name string, namespace string) (*model.Pipeline, error) {
-	return r.pipelineStore.GetPipelineByNameAndNamespace(name, namespace)
+	if pipeline, err := r.pipelineStore.GetPipelineByNameAndNamespace(name, namespace); err != nil {
+		return nil, util.Wrapf(err, "Failed to get a pipeline named %v in namespace %v", name, namespace)
+	} else {
+		return pipeline, nil
+	}
 }
 
+// TODO(gkcalat): consider removing after KFP v2 GA if users are not affected.
+// Returns a pipeline specified by name and namespace using LEFT JOIN on SQL query.
+// This could be more performant for a large number of pipeline versions.
+func (r *ResourceManager) GetPipelineByNameAndNamespaceV1(name string, namespace string) (*model.Pipeline, *model.PipelineVersion, error) {
+	if pipeline, pipelineVersion, err := r.pipelineStore.GetPipelineByNameAndNamespaceV1(name, namespace); err != nil {
+		return nil, nil, util.Wrapf(err, "ResourceManager (v1beta1): Failed to get a pipeline named %v in namespace %v", name, namespace)
+	} else {
+		return pipeline, pipelineVersion, nil
+	}
+}
+
+// Deletes a pipeline. Does not delete pipeline spec in the object storage.
+// Fails if the pipeline has existing pipeline versions.
 func (r *ResourceManager) DeletePipeline(pipelineId string) error {
+	// Check if pipeline exists
 	_, err := r.pipelineStore.GetPipeline(pipelineId)
 	if err != nil {
-		return util.Wrap(err, "Delete pipeline failed")
+		return util.Wrapf(err, "Failed to delete pipeline with id %v as it was not found", pipelineId)
+	}
+
+	// if apiVersion == "v1beta1" {
```

Comment:
Need to handle the commented code here.

Reply:
This is deprecated. I removed the commented code. We no longer allow users to delete pipelines with existing pipeline versions in both v1 and v2.
/cc @chensun